rir/src/compiler/native/lower_function_llvm.cpp

```diff
@@ -1691,7 +1700,10 @@ void LowerFunctionLLVM::compileBinop(
 
     builder.SetInsertPoint(done);
     if (rep == Rep::SEXP) {
-        setVal(i, box(res(), lhs->type.mergeWithConversion(rhs->type), false));
+        auto to = lhs->type.mergeWithConversion(rhs->type);
+        if (intIntToReal)
+            to = to.notT(RType::integer).orT(RType::real);
```

Comment:
@o- not sure this is the best way.. need to box as real but both lhs and rhs are ints

Reply:
sounds good to me, R agrees

```
> typeof(1L/1L)
[1] "double"
```

Reply:
wait! actually shouldnt this be `box(res, i->type, false)`?  I mean the type of `i` should really be `real` in this case, otherwise we will surely have more problems still. I am actually afraid that we made this same bug in type inference too...

Reply:
Ah, right.. the type of Div is `(int|real)` though.. so then the inference indeed must figure out that it in fact should be real only

Reply:
right, ic. not sure which one is the better solution then...